### DISCUSSION TITLE
Adding support for raw in vips_target

### DISCUSSION THREAD

**Discussion by akash-akya** [libvips/libvips#3940] [Ideas]
Does it makes sense to create `VipsForeignSaveRawTarget` to support streaming raw pixels in a unified way (`vips_image_write_to_target(in, ".raw", target)`)?
I know there is already `VipsForeignSaveRawFd` which can be used to create similar interface, but in our case we have a unified interface to do streaming writes and this would help us to avoid creating special case to write raw pixels.

If this makes sense then I can try implementing it

**Comment by jcupitt** [comment]
Hi @akash-akya,

Sure, this sounds like a good idea. You can copy-paste code from something like `ppmsave`. I would have an abstract base class `VipsForeignSaveRaw` which writes to a target, then define `VipsForeignSaveRawTarget`, `VipsForeignSaveRawFile` etc. as concrete subclasses which define the parameters.

Let's make an enhancement issue for this feature.

**Reply by jcupitt** [reply]
Enhanceemnt issue here: https://github.com/libvips/libvips/issues/3941

Let's continue any discussion on that.
